src/mlpack/core/data/load_impl.hpp

```diff
@@ -77,51 +77,28 @@ bool inline inplace_transpose(arma::Mat<eT>& X)
   }
 }
 
-template<typename eT>
-bool Load(const std::string& filename,
-          arma::Mat<eT>& matrix,
-          const bool fatal,
-          const bool transpose)
+inline
+std::string AutoDetect(std::fstream& stream,
+                      const std::string filename,
+                      arma::file_type& detectedLoadType,
+                      const bool fatal)
 {
-  Timer::Start("loading_data");
-
   // Get the extension.
   std::string extension = Extension(filename);
-
-  // Catch nonexistent files by opening the stream ourselves.
-  std::fstream stream;
-#ifdef  _WIN32 // Always open in binary mode on Windows.
-  stream.open(filename.c_str(), std::fstream::in | std::fstream::binary);
-#else
-  stream.open(filename.c_str(), std::fstream::in);
-#endif
-  if (!stream.is_open())
-  {
-    Timer::Stop("loading_data");
-    if (fatal)
-      Log::Fatal << "Cannot open file '" << filename << "'. " << std::endl;
-    else
-      Log::Warn << "Cannot open file '" << filename << "'; load failed."
-          << std::endl;
-
-    return false;
-  }
-
+  std::string stringType = "";
   bool unknownType = false;
-  arma::file_type loadType;
-  std::string stringType;
 
   if (extension == "csv" || extension == "tsv")
   {
-    loadType = arma::diskio::guess_file_type(stream);
-    if (loadType == arma::csv_ascii)
+    detectedLoadType = arma::diskio::guess_file_type(stream);
```

Comment:
Please do not use the `arma::diskio::guess_file_type()` function. It's an internal armadillo function, subject to change and/or removal.  It was never designed or intended to be used by user code.

Only the functions officially documented are considered as publicly accessible. Extract from armadillo [documetation](http://arma.sourceforge.net/docs.html#api_additions) on the use of internal functions:

> Caveat: any function, class, constant or other code not explicitly described in the public API documentation is considered as part of the underlying internal implementation details, and may change or be removed without notice. (In other words, don't use internal functionality). 

I suggest writing a dedicated mlpack function to accomplish the detection of the file type.  It's possible to adapt the current code from `arma::diskio::guess_file_type()` and improve upon it.

Reply:
Thanks for review. Working on it.